content/2016.09.trying-kinto.md

```diff
@@ -0,0 +1,420 @@
+Le résultat est disponible sur [jedit.org/trackers](http://jedit.org/trackers/). Pour limiter le volume, seuls les
+tickets ouverts sont disponibles.
+
+Je lance la moulinette à peu près toutes les semaines (elle a besoin de mes droits d'admin pour récupérer l'archive).
+Je déploie ensuite avec rsync les fichiers HTML et un tar.
+
+La génération de l'archive du projet par Sourceforge prend plusieurs minutes, c'est pourquoi je ne la lance pas très souvent.
+Pour les utilisateurs, ce n'est pas très pratique, car il faut qu'ils téléchargent une nouvelle archive pour avoir une
+vue à jour des tickets. Elle peut avoir une semaine de retard : peut mieux faire.
+
+**Objectifs**
+
+ - je voudrais permettre la mise à jour incrémentale des tickets.
+ - je voudrais enrichir la visualisation des tickets : possibilité de recherche améliorée,
+   peut-être permettre la modification hors-ligne.
+ - je voudrais essayer Kinto :-).
+
+# Mes premiers pas avec Kinto
+
+J'ai entendu parler de Kinto par Remy et Alexis. Comme avec couchdb/pouchdb, il est possible de synchroniser
+la base de données entre le serveur (Kinto) et le navigateur web (kinto.js). Kinto.js stocke les données dans IndexedDB, donc il est
+possible de manipuler une base relativement volumineuse.
+
+Chaque utilisateur peut créer ses propres collections (l'équivalent des tables SQL) Kinto. Il est même possible de gérer les
+permissions document par document (l'équivalent des lignes SQL).
+
+**Le tutoriel : ToDo list**
+
+J'ai commencé par le
+[tutoriel](http://kintojs.readthedocs.io/en/latest/tutorial/). Il présente bien les opérations: créer des documents, les
+afficher, les modifier, synchroniser avec le serveur. Il s'appuie sur le serveur Kinto de test de mozilla : donc un
+obstacle de moins pour démarrer.
+
+**Des tickets au lieu des tâches**
+
+Je suis passé à un serveur kinto local pour mes tests: il est déployable très facilement via docker.
+Si vous envisagez d'importer des documents, utilisez tout de suite le backend PostgreSQL
+(aussi facile à mettre en place, avec docker-compose ;
+tout est [expliqué](http://kinto.readthedocs.io/en/stable/tutorials/install.html#using-docker-compose) ).
+Le backend en RAM pose des [problèmes](https://github.com/Kinto/kinto/issues/759) lors d'import batch en parallèle (ce
+qui se produit si vous importez depuis un [script](https://github.com/elelay/jedit-trackers-kinto/blob/master/transform.js)
+node.js utilisant [request](https://www.npmjs.com/package/request)).
+
+Une fois les tickets importés, l'application web n'a plus qu'à les récupérer via la synchronisation kinto.js et les
+afficher.
+
+**Recherche avec kinto.js**
+
+La recherche avec kinto.js est assez rudimentaire : plusieurs `clé: [valeur1, valeur2, ...]`. Seul le premier critère a une chance d'être
+optimisé, si j'ai bien suivi le [code](https://github.com/Kinto/kinto.js/blob/master/src/adapters/IDB.js#L325). Ensuite,
```

Comment:
C'est une bonne idée de ne pas linker sur master, car ce dernier va inévitablement changer au fil du temps. Ici, un lien vers `https://github.com/Kinto/kinto.js/blob/f48ea470ddb2c6398acc386439043e6cae8bffe4/src/adapters/IDB.js#L325` sera pérenne.